### PR TITLE
Tictactoe

### PR DESCRIPTION
**Pull request by rida-elbahtouri**
## In this branch we :
- Created the HTML side of the game Tic-Tac-Toe
- build the logic using javascript
- made the players capable of restart matches

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @rida-elbahtouri :clap: However, some requirements need to be implemented in order to get approval for this milestone. Check :arrow_down: 

#### Behaviour (required) :warning: 
* [x] It's impossible to restart a game after a draw. The restart message keeps popping after clicking a cell  in the board.
- I noticed some spelling errors in the messages displayed. kindly fix them

#### General (required) :warning: 
* [x] No `console.log` commited to repo. It's ugly, it kills performance. and it can make confidential data be visible in the browser tools. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/console_log_requirement.png) from Pathwright)
* [x] ES6 - using object destructuring to get the values from an object. This way you can avoid repeating a lot of code. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/destructuring_requirement.png) from Pathwright)
* [x] Keep the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)
* [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)
#### Implementation (required) :warning: 
* [x] It doesn't use `global` code - encapsulate logic inside modules or factories ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/global_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
* [x] It has a function with the only responsibility of rendering the contents of the gameboard array to the webpage ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/render_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
- Keep the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)
### That's it :relaxed: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: main.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    } else {
+      return board;
+    }
+  };
+
+  const validateMove = (cells, players) => {
+    let player = 0;
+    cells.forEach((cell) => {
+      cell.addEventListener("click", function () {
+        if (cell.innerHTML.length == 0) {
```
Comment (whiz25) [inline main.js:69]: - any piece of code related to the view (DOM) should not be here. You need to move this code into the dom.js file. This was already pointed out in the previous review.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+startGame.addEventListener("click", function (e) {
+  e.preventDefault();
+  let players_lis = [];
+  let board = GameBoard();
+  players_lis.push(Player(players[0].value, "X", 0));
+  players_lis.push(Player(players[1].value, "O", 0));
+  movehtml.innerHTML = `hello players , ${players_lis[0].name} will play with ${players_lis[0].mark} 
+  and ${players_lis[1].name} will play with ${players_lis[1].mark}`;
+  movehtml.style.padding = "10px";
```
Comment (whiz25) [inline main.js:114]: - this should be in the DOM file. This file should only contain pure JS and nothing more

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray:
Nice work implementing changes required by the previous reviewer. There is however, a few things we need to fix. There is still logic concerning the DOM in the main.js file. Please separate the view logic from the JS logic.

### Martin Afani

**Comment by rida-elbahtouri** [comment]
### Note:
I couldn't move the validateMove function to dom.js, because it depends on variables that exist in the main.js file, and If I pass them as parameters that will cause problems with the logic, the game will refuse to restart

**Review by whiz25 — APPROVED** [review]
## Approved :clap: 
Merge and proceed to the next milestone.

### Martin Afani

**Pull request merged**